src/client/Item/Item.ts

```diff
@@ -4,17 +4,50 @@ import { RequestOptions, Response } from "../../request";
 import { Container } from "../Container";
 import { ItemResponse } from "./ItemResponse";
 
+/**
+ * Used to perform operations on a specific item.
+ *
+ * @see {@link Items} for operations on all items; see `container.items`.
+ */
 export class Item {
   private client: CosmosClient;
+  /**
+   * Returns a reference URL to the resource. Used for linking in Permissions.
+   */
   public get url() {
     return UriFactory.createDocumentUri(this.container.database.id, this.container.id, this.id);
   }
 
+  /**
+   * @hidden
+   * @param container The parent {@link Container}.
+   * @param id The id of the given {@link Item}.
+   * @param primaryKey The primary key of the given {@link Item} (only for partitioned containers).
+   */
   constructor(public readonly container: Container, public readonly id: string, public readonly primaryKey: string) {
     this.client = this.container.database.client;
   }
 
+  /**
+   * Read the item's definition.
+   *
+   * There is no set schema for JSON items. They may contain any number of custom properties.
+   *
+   * @param options Additional options for the request, such as the partition key.
+   * Note, if you provide a partition key here, it will override the primary key on this object.
+   */
   public read(options?: RequestOptions): Promise<ItemResponse<any>>;
+  /**
+   * Read the item's definition.
+   *
+   * Any provided type, T, is not necessarily enforced by the SDK.
```

Comment:
Do we provide an example of this anywhere? Not sure it needs to be inline, but could be linked from these methods.

Reply:
I provided 1 example, but I think when we have proper docs, we can have a link to a section on our generics support. Our support for generic is kinda flakey, since we really only support interfaces. Classes will "work", but if you "typeof" our response, it will always be "object".

Reply:
We might be able to add a feature where you pass a class which we'll new up along on the options object, but that's a separate issue.

Reply:
That would be cool!